sci-libs/openblas/openblas-0.3.5.ebuild

```diff
@@ -0,0 +1,76 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+DESCRIPTION="Optimized BLAS library based on GotoBLAS2"
+HOMEPAGE="http://xianyi.github.com/OpenBLAS/"
+SRC_URI="https://github.com/xianyi/OpenBLAS/tarball/v${PV} -> ${P}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86 ~amd64-linux ~x86-linux ~x64-macos ~x86-macos"
+IUSE="dynamic openmp pthread serial static-libs eselect-ldso"
+REQUIRED_USE="?? ( openmp pthread serial )"
+
+RDEPEND="
+>=app-eselect/eselect-blas-0.2
+>=app-eselect/eselect-lapack-0.2
+"
+DEPEND="${RDEPEND}
+	virtual/pkgconfig"
+
+PATCHES=( "${FILESDIR}/shared-blas-lapack.patch" )
+
+openblas_flags() {
+	local flags=()
+	use dynamic && \
+		flags+=( DYNAMIC_ARCH=1 TARGET=GENERIC NUM_THREADS=64 NO_AFFINITY=1 )
+	if use openmp; then
+		flags+=( USE_THREAD=1 USE_OPENMP=0 )
+	elif use pthread; then
+		flags+=( USE_THREAD=1 USE_OPENMP=1 )
```

Comment:
Is this logic correct?

Reply:
This logic is correct due to the weird upstream makefile logic.

Reply:
```